lib/iris/tests/unit/cube/test_Cube__aggregated_by.py

```diff
@@ -0,0 +1,516 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""Unit tests for the `iris.cube.Cube` class aggregated_by method."""
+
+# import iris tests first so that some things can be initialised
+# before importing anything else.
+import iris.tests as tests  # isort:skip
+
+from unittest import mock
+
+from cf_units import Unit
+import numpy as np
+
+from iris._lazy_data import as_lazy_data
+import iris.analysis
+from iris.analysis import MEAN, Aggregator
+import iris.aux_factory
+import iris.coords
+from iris.coords import AncillaryVariable, AuxCoord, CellMeasure, DimCoord
+from iris.cube import Cube
+import iris.exceptions
+
+
+class Test_aggregated_by(tests.IrisTest):
```

Comment:
None of the tests in this file appear to be running in the CI and I don't know why

Reply:
Does CI rely on the ifmain?  E.g.

https://github.com/wjbenfold/iris/blob/239114f89118573880a1d4fa0799f2e9af8a4bd7/lib/iris/tests/unit/cube/test_Cube.py#L2878-L2879

Reply:
I should probably have one anyway, I've added it

Reply:
Still not finding them

Reply:
Still not finding them

Reply:
I think I've hunted this down - the file was executable, which is apparently enough for the test runner to decide it can't possibly be a test

Reply:
🤯 

Good hunting skills!

Reply:
"Once you eliminate every line in the file and the filename, whatever difference remains must be the bug" - Sherlock Holmes